lib/archive.ts

```diff
@@ -3,57 +3,36 @@ import * as path from 'path';
 import * as crypto from 'crypto';
 import * as fs from 'fs';
 import * as glob from 'glob';
-import * as needle from 'needle';
```

Comment:
can we remove needle as a dependency now?

Reply:
As far as I can see, the http client returns 

```{
  res: NeedleResponse;
  body: unknown;
}```

and we also use `method: NeedleHttpVerbs;` in the request. Not sure about typescript stuff, but I guess I could just define `HttpResponse { statusCode: int} ` and get do the same for the verb and it should work. Is that what you had in mind?

Reply:
if we only use the types, can it be moved to a devDependency?